src/Shopsys/ShopBundle/Model/Administrator/Security/AdministratorUserProvider.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Shopsys\ShopBundle\Model\Administrator\Security;
+
+use DateTime;
+use Shopsys\FrameworkBundle\Model\Administrator\Administrator;
+use Shopsys\FrameworkBundle\Model\Administrator\Security\AdministratorUserProvider as BaseAdministratorUserProvider;
+use Shopsys\FrameworkBundle\Model\Security\TimelimitLoginInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class AdministratorUserProvider extends BaseAdministratorUserProvider
+{
+    /**
+     * @param \Symfony\Component\Security\Core\User\UserInterface $userInterface
+     * @return \Shopsys\FrameworkBundle\Model\Administrator\Administrator
+     */
+    public function refreshUser(UserInterface $userInterface)
+    {
+        $class = get_class($userInterface);
+        if (!$this->supportsClass($class)) {
+            $message = sprintf('Instances of "%s" are not supported.', $class);
+            throw new \Symfony\Component\Security\Core\Exception\UnsupportedUserException($message);
+        }
+
+        /** @var \Shopsys\FrameworkBundle\Model\Administrator\Administrator $administrator */
+        $administrator = $userInterface;
+
+        $freshAdministrator = $this->administratorRepository->findById($administrator->getId());
+
+        if ($administrator instanceof TimelimitLoginInterface) {
+            if (time() - $administrator->getLastActivity()->getTimestamp() > 3600 * 5) {
+                throw new \Symfony\Component\Security\Core\Exception\AuthenticationExpiredException('Admin was too long inactive.');
+            }
+            if ($freshAdministrator !== null) {
+                $freshAdministrator->setLastActivity(new DateTime());
+            }
+        }
+
+        if ($freshAdministrator === null) {
```

Comment:
This should be checked right after variable assigning. Then you can remove condition above.

Reply:
Sorry I do not want to do this change, because when I am extending some class from framework, I try to do as minimal changes as possible so people that will be upgrading demoshop in the future have as little problems as possible. Diff is easier etc.